src/matrix_multiply_omp_compare/CMakeLists.txt

```diff
@@ -1,44 +1,23 @@
-############################################################################
-#
-#  Copyright (C) Codeplay Software Limited
-#  Licensed under the Apache License, Version 2.0 (the "License");
-#  you may not use this file except in compliance with the License.
-#  You may obtain a copy of the License at
-#
-#      http://www.apache.org/licenses/LICENSE-2.0
-#
-#  Unless required by applicable law or agreed to in writing, software
-#  distributed under the License is distributed on an "AS IS" BASIS,
-#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-#  See the License for the specific language governing permissions and
-#  limitations under the License.
-#
-############################################################################
+add_executable(matrix_multiply_omp_compare matrix-multiply.cpp)
+set(MATRIX_MULTIPLY_FLAGS ${SYCL_FLAGS})
 
-# Set the project name
-project(SYCL-samples_matrix_multiply_omp_compare
-        VERSION 1.0
-        DESCRIPTION "Compares an OpenMP blocked matrix multiplication 
-        implementation with a SYCL blocked matrix multiplication example"
-        LANGUAGES CXX)
+find_package(OpenMP QUIET)
+if(OpenMP_CXX_FOUND)
+    target_link_libraries(matrix_multiply_omp_compare PUBLIC OpenMP::OpenMP_CXX)
+elseif(CMAKE_CXX_COMPILER MATCHES "clang")
+    # CMake's FindOpenMP doesn't recognise LLVM internal implementation, but
+    # it's still possible to parallelise OpenMP loops with clang++ -fopenmp
+    list(APPEND MATRIX_MULTIPLY_FLAGS -fopenmp)
+    if(CMAKE_BUILD_TYPE STREQUAL "")
```

Comment:
This is a great explanation and maybe we should promote it all the way to the top level CMake - if the build type is unset, make it Release? We want the demos to run fast, and if the user wants debug they can opt-in for debugging or similar.

Reply:
I later realised that this comment was actually untrue and removed this section, but forgot to save the file and amend the commit. :man_facepalming: There is actually a difference between `clang++ -O0` and `clang++ -fopenmp -O0`, the latter is still faster.

Nevertheless, the issue of unset build type still holds, so I just did as you suggested in the top-level config.